<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interoperation with other tools • massdataset</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Interoperation with other tools">
<meta property="og:description" content="massdataset">
<meta property="og:image" content="https://tidymass.github.io/massdataset/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-000000-01"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-000000-01');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">massdataset</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.27</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Help documents
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">Installation</li>
    <li>
      <a href="../articles/massdataset_install.html">Install massdataset</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">mass_dataset-class object</li>
    <li>
      <a href="../articles/data_import_and_export.html">Data import and export</a>
    </li>
    <li>
      <a href="../articles/mutate_ms2.html">Add MS2 spectra data into mass_dataset class object</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Information from mass_dataset</li>
    <li>
      <a href="../articles/extract_data.html">Extract data from mass_dataset</a>
    </li>
    <li>
      <a href="../articles/process_info.html">Processing information</a>
    </li>
    <li>
      <a href="../articles/html_process_info.html">Output html processing information</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Processing of mass_dataset</li>
    <li>
      <a href="../articles/summary_info.html">Summary information</a>
    </li>
    <li>
      <a href="../articles/add_information.html">Add new information to mass_dataset</a>
    </li>
    <li>
      <a href="../articles/filtering.html">Filtering samples/variables</a>
    </li>
    <li>
      <a href="../articles/accessor.html">Accessor</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Merge and split mass_dataset</li>
    <li>
      <a href="../articles/cbind_rbind_mass_dataset.html">rbind or cbind mass_dataset</a>
    </li>
    <li>
      <a href="../articles/merge_two_mass_dataset.html">Merge two mass_dataset</a>
    </li>
    <li>
      <a href="../articles/split_mass_dataset.html">Split mass_dataset</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Others</li>
    <li>
      <a href="../articles/tidyverse_verse.html">Process mass_dataset using tidyverse functions</a>
    </li>
    <li>
      <a href="../articles/base_function.html">R base functions for mass_dataset</a>
    </li>
    <li>
      <a href="../articles/ggplot_mass_dataset.html">mass_dataset and ggplot2</a>
    </li>
    <li>
      <a href="../articles/based_on_mass_dataset.html">Develop functions based on mass_dataset</a>
    </li>
    <li>
      <a href="../articles/interoperability_with_other_tools.html">Interoperation with other tools</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Blog posts</li>
    <li>
      <a href="../articles/massdataset.html">Version 0.4.0</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../news/index.html">Change log</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/contact_us.html">Contact us</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/tidymass/massdataset/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/xiaotaoshen1990" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interoperation with other tools</h1>
                        <h4 data-toc-skip class="author">Xiaotao Shen
(<a href="https://www.shenxt.info/" class="external-link uri">https://www.shenxt.info/</a>)</h4>
                        
            <h4 data-toc-skip class="date">Created on 2022-05-16 and
updated on 2023-10-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/tidymass/massdataset/blob/HEAD/vignettes/interoperability_with_other_tools.Rmd" class="external-link"><code>vignettes/interoperability_with_other_tools.Rmd</code></a></small>
      <div class="hidden name"><code>interoperability_with_other_tools.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">
<strong>Introduction</strong><a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>To make <code>tidyMass</code> and <code>massdataset</code> is more
interoperability with other toolts which have beed developed for omics
data processing and analysis, we provide several functions that could
convert “mass_dataset” and data formats that required by other tools. In
the meanwhile, functions that convert other data formats to
<code>mass_dataset</code> are also provided.</p>
</div>
<div class="section level2">
<h2 id="metdna">
<strong>MetDNA</strong><a class="anchor" aria-label="anchor" href="#metdna"></a>
</h2>
<p><code>MetDNA</code> is a web-based tool for metabolite annotation
using metabolic reaction network (<a href="http://metdna.zhulab.cn/" class="external-link uri">http://metdna.zhulab.cn/</a>). Users can eaisy convert
<code>mass_dataset</code> to files that required for
<code>MetDNA</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tidymass/massdataset" class="external-link">massdataset</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"expression_data"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sample_info"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sample_info_note"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"variable_info"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"variable_info_note"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">object</span> <span class="op">=</span></span>
<span>  <span class="fu"><a href="../reference/create_mass_dataset.html">create_mass_dataset</a></span><span class="op">(</span></span>
<span>    expression_data <span class="op">=</span> <span class="va">expression_data</span>,</span>
<span>    sample_info <span class="op">=</span> <span class="va">sample_info</span>,</span>
<span>    variable_info <span class="op">=</span> <span class="va">variable_info</span>,</span>
<span>    sample_info_note <span class="op">=</span> <span class="va">sample_info_note</span>,</span>
<span>    variable_info_note <span class="op">=</span> <span class="va">variable_info_note</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">object</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">-------------------- </span></span></span>
<span><span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">massdataset version: 1.0.12 </span></span></span>
<span><span class="co"><span style="color: #00BB00;">#&gt; </span><span style="color: #BBBB00;">-------------------- </span></span></span>
<span><span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">1.expression_data:</span>[ 1000 x 8 data.frame]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">2.sample_info:</span>[ 8 x 4 data.frame]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">3.variable_info:</span>[ 1000 x 3 data.frame]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">4.sample_info_note:</span>[ 4 x 2 data.frame]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">5.variable_info_note:</span>[ 3 x 2 data.frame]</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">6.ms2_data:</span>[ 0 variables x 0 MS2 spectra]</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">-------------------- </span></span></span>
<span><span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">Processing information (extract_process_info())</span></span></span>
<span><span class="co"><span style="color: #00BB00;">#&gt; 1 processings in total</span></span></span>
<span><span class="co"><span style="color: #00BB00;">#&gt; create_mass_dataset ---------- </span></span></span>
<span><span class="co"><span style="color: #00BB00;">#&gt; </span>      Package         Function.used                Time</span></span>
<span><span class="co">#&gt; 1 massdataset create_mass_dataset() 2022-08-07 19:35:20</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/export_mass_dataset4metdna.html">export_mass_dataset4metdna</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span>, </span>
<span>                           path <span class="op">=</span> <span class="st">"convert/metdna"</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>The files will be exported in the folder “convert/metdna”.</p>
<p><img src="../reference/figures/Screen-Shot.png"></p>
<p>Peak table.</p>
<p><img src="../reference/figures/Screen-Shot%201.png"> sample_info.</p>
<p><img src="../reference/figures/Screen-Shot%202.png"></p>
</div>
<div class="section level2">
<h2 id="summarizedexperiment">
<strong>SummarizedExperiment</strong><a class="anchor" aria-label="anchor" href="#summarizedexperiment"></a>
</h2>
<p>The <code>SummarizedExperiment</code> class is used to store
rectangular matrices of experimental results, which are commonly
produced by sequencing and microarray experiments. This data structure
is supported by lots of tools in omics files in R environment. We can
use the <code>convert_mass_dataset2summarizedexperiment</code> function
to convert <code>mass_dataset</code> to
<code>SummarizedExperiment</code> class.</p>
<p>Please install <code>SummarizedExperiment</code> first.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">BiocManager</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SummarizedExperiment"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="va">se_object</span> <span class="op">&lt;-</span> </span>
<span><span class="fu"><a href="../reference/convert_mass_dataset2summarizedexperiment.html">convert_mass_dataset2summarizedexperiment</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="va">se_object</span></span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 1000 8 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(1000): M136T55_2_POS M79T35_POS ... M232T937_POS M301T277_POS</span></span>
<span><span class="co">#&gt; rowData names(3): variable_id mz rt</span></span>
<span><span class="co">#&gt; colnames(8): Blank_3 Blank_4 ... PS4P3 PS4P4</span></span>
<span><span class="co">#&gt; colData names(4): sample_id injection.order class group</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="mztab-m-format">
<strong>mzTab-m format</strong><a class="anchor" aria-label="anchor" href="#mztab-m-format"></a>
</h2>
<p><code>mzTab-M</code> is a data standard for sharing quantitative
results in mass spectrometry metabolomics, which is also supported by
lots of tools in metabolomics/proteomics filed (<a href="https://pubs.acs.org/doi/10.1021/acs.analchem.8b04310" class="external-link uri">https://pubs.acs.org/doi/10.1021/acs.analchem.8b04310</a>).
In <code>massdataset</code>, we also provide two function to convert
<code>mass_data</code> class and <code>mzTab-m</code>.</p>
<div class="section level3">
<h3 id="convert-mass_dataset-to-mztab-m">Convert <code>mass_dataset</code> to <code>mzTab-M</code><a class="anchor" aria-label="anchor" href="#convert-mass_dataset-to-mztab-m"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/convert_mass_dataset2mztab.html">convert_mass_dataset2mztab</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span>, </span>
<span>                           path <span class="op">=</span> <span class="st">"convert/mztab"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>The data is put in the folder “convert/mztab”. You can open it with
Excel.</p>
<p><img src="../reference/figures/Screen-Shot%203.png"></p>
</div>
</div>
<div class="section level2">
<h2 id="rformassspectrometry">
<em>RforMassSpectrometry</em><a class="anchor" aria-label="anchor" href="#rformassspectrometry"></a>
</h2>
<p><a href="https://www.rformassspectrometry.org/" class="external-link">RforMassSpectrometry</a> is
a project that contains several R software for the analysis and
interpretation of high throughput mass spectrometry assays. We can
eaisly convert <code>mass_dataset</code> to the format that it require
and then analysis using RforMassSpectrometry. Next, we will give an
example how to use the <code>MetaboAnnotation</code> in
RforMassSpectrometry for annotation.</p>
<p>Please install <code>MetaboAnnotation</code> first.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">BiocManager</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">MetaboAnnotation</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MetaboAnnotation"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">MetaboAnnotation</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span></code></pre></div>
<p>Convert <code>mass_dataset</code> class to
<code>SummarizedExperiment</code> class object.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_object</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/convert_mass_dataset2summarizedexperiment.html">convert_mass_dataset2summarizedexperiment</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_object</span> </span>
<span><span class="co">#&gt; class: SummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 1000 8 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(1000): M136T55_2_POS M79T35_POS ... M232T937_POS M301T277_POS</span></span>
<span><span class="co">#&gt; rowData names(3): variable_id mz rt</span></span>
<span><span class="co">#&gt; colnames(8): Blank_3 Blank_4 ... PS4P3 PS4P4</span></span>
<span><span class="co">#&gt; colData names(4): sample_id injection.order class group</span></span></code></pre></div>
<p>Get the targeted table (database)</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_df</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"LipidMaps_CompDB.txt"</span>,</span>
<span>                package <span class="op">=</span> <span class="st">"MetaboAnnotation"</span><span class="op">)</span>,</span>
<span>    header <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    sep <span class="op">=</span> <span class="st">"\t"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">target_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;   headgroup        name exactmass    formula chain_type</span></span>
<span><span class="co">#&gt; 1       NAE  NAE 20:4;O  363.2773  C22H37NO3       even</span></span>
<span><span class="co">#&gt; 2       NAT  NAT 20:4;O  427.2392 C22H37NO5S       even</span></span>
<span><span class="co">#&gt; 3       NAE NAE 20:3;O2  381.2879  C22H39NO4       even</span></span>
<span><span class="co">#&gt; 4       NAE    NAE 20:4  347.2824  C22H37NO2       even</span></span>
<span><span class="co">#&gt; 5       NAE    NAE 18:2  323.2824  C20H37NO2       even</span></span>
<span><span class="co">#&gt; 6       NAE    NAE 18:3  321.2668  C20H35NO2       even</span></span></code></pre></div>
<p>We need to change the column names to make it fit to
<code>MetaboAnnotation</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">se_object</span><span class="op">)</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/mass_dataset-extracting.html">extract_variable_info</a></span><span class="op">(</span><span class="va">object</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>feature_id <span class="op">=</span> <span class="va">variable_id</span>,</span>
<span>                rtime <span class="op">=</span> <span class="va">rt</span><span class="op">)</span></span></code></pre></div>
<p>Metabolite annotation.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parm</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Mass2MzParam</span><span class="op">(</span></span>
<span>    adducts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"[M+H]+"</span>, <span class="st">"[M+Na]+"</span><span class="op">)</span>,</span>
<span>    tolerance <span class="op">=</span> <span class="fl">0.005</span>,</span>
<span>    ppm <span class="op">=</span> <span class="fl">0</span></span>
<span>  <span class="op">)</span></span>
<span><span class="va">matched_features</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">matchValues</span><span class="op">(</span><span class="va">se_object</span>, <span class="va">target_df</span>, param <span class="op">=</span> <span class="va">parm</span><span class="op">)</span></span>
<span><span class="va">matched_features</span></span>
<span><span class="co">#&gt; Object of class Matched </span></span>
<span><span class="co">#&gt; Total number of matches: 455 </span></span>
<span><span class="co">#&gt; Number of query objects: 1000 (217 matched)</span></span>
<span><span class="co">#&gt; Number of target objects: 57599 (375 matched)</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">matchedData</span><span class="op">(</span><span class="va">matched_features</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 1238 rows and 11 columns</span></span>
<span><span class="co">#&gt;                   feature_id        mz     rtime target_headgroup target_name</span></span>
<span><span class="co">#&gt;                  &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;      &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; M136T55_2_POS  M136T55_2_...  136.0614   54.9790               NA          NA</span></span>
<span><span class="co">#&gt; M79T35_POS        M79T35_POS   79.0539   35.3655               NA          NA</span></span>
<span><span class="co">#&gt; M307T548_POS   M307T548_P...  307.1403  547.5664               NA          NA</span></span>
<span><span class="co">#&gt; M183T224_POS   M183T224_P...  183.0621  224.3278               NA          NA</span></span>
<span><span class="co">#&gt; M349T47_POS    M349T47_PO...  349.0158   47.0026               NA          NA</span></span>
<span><span class="co">#&gt; ...                      ...       ...       ...              ...         ...</span></span>
<span><span class="co">#&gt; M343T707_2_POS M343T707_2...   343.339   707.297               NA          NA</span></span>
<span><span class="co">#&gt; M236T543_POS   M236T543_P...   236.172   542.567               NA          NA</span></span>
<span><span class="co">#&gt; M232T937_POS   M232T937_P...   231.986   936.608               NA          NA</span></span>
<span><span class="co">#&gt; M301T277_POS   M301T277_P...   301.200   277.116               FA  FA 16:2;O3</span></span>
<span><span class="co">#&gt; M301T277_POS   M301T277_P...   301.200   277.116               MG   MG 13:2;O</span></span>
<span><span class="co">#&gt;                target_exactmass target_formula target_chain_type      adduct</span></span>
<span><span class="co">#&gt;                       &lt;numeric&gt;    &lt;character&gt;       &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt; M136T55_2_POS                NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M79T35_POS                   NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M307T548_POS                 NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M183T224_POS                 NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M349T47_POS                  NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; ...                         ...            ...               ...         ...</span></span>
<span><span class="co">#&gt; M343T707_2_POS               NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M236T543_POS                 NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M232T937_POS                 NA             NA                NA          NA</span></span>
<span><span class="co">#&gt; M301T277_POS            300.194       C16H28O5              even      [M+H]+</span></span>
<span><span class="co">#&gt; M301T277_POS            300.194       C16H28O5               odd      [M+H]+</span></span>
<span><span class="co">#&gt;                       score ppm_error</span></span>
<span><span class="co">#&gt;                   &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; M136T55_2_POS            NA        NA</span></span>
<span><span class="co">#&gt; M79T35_POS               NA        NA</span></span>
<span><span class="co">#&gt; M307T548_POS             NA        NA</span></span>
<span><span class="co">#&gt; M183T224_POS             NA        NA</span></span>
<span><span class="co">#&gt; M349T47_POS              NA        NA</span></span>
<span><span class="co">#&gt; ...                     ...       ...</span></span>
<span><span class="co">#&gt; M343T707_2_POS           NA        NA</span></span>
<span><span class="co">#&gt; M236T543_POS             NA        NA</span></span>
<span><span class="co">#&gt; M232T937_POS             NA        NA</span></span>
<span><span class="co">#&gt; M301T277_POS   -0.000767133   2.54691</span></span>
<span><span class="co">#&gt; M301T277_POS   -0.000767133   2.54691</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-information">
<strong>Session information</strong><a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS Big Sur ... 10.16</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib</span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] MetaboAnnotation_1.0.0      SummarizedExperiment_1.26.1</span></span>
<span><span class="co">#&gt;  [3] Biobase_2.56.0              GenomicRanges_1.48.0       </span></span>
<span><span class="co">#&gt;  [5] GenomeInfoDb_1.32.2         IRanges_2.30.0             </span></span>
<span><span class="co">#&gt;  [7] S4Vectors_0.34.0            BiocGenerics_0.42.0        </span></span>
<span><span class="co">#&gt;  [9] MatrixGenerics_1.8.1        matrixStats_0.62.0         </span></span>
<span><span class="co">#&gt; [11] BiocManager_1.30.18         forcats_0.5.1.9000         </span></span>
<span><span class="co">#&gt; [13] stringr_1.4.0               dplyr_1.0.9                </span></span>
<span><span class="co">#&gt; [15] purrr_0.3.4                 readr_2.1.2                </span></span>
<span><span class="co">#&gt; [17] tidyr_1.2.0                 tibble_3.1.7               </span></span>
<span><span class="co">#&gt; [19] tidyverse_1.3.1             ggplot2_3.3.6              </span></span>
<span><span class="co">#&gt; [21] magrittr_2.0.3              masstools_1.0.2            </span></span>
<span><span class="co">#&gt; [23] massdataset_1.0.12         </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] readxl_1.4.0                backports_1.4.1            </span></span>
<span><span class="co">#&gt;   [3] circlize_0.4.15             systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt;   [5] igraph_1.3.2                plyr_1.8.7                 </span></span>
<span><span class="co">#&gt;   [7] lazyeval_0.2.2              BiocParallel_1.30.3        </span></span>
<span><span class="co">#&gt;   [9] Rdisop_1.56.0               digest_0.6.29              </span></span>
<span><span class="co">#&gt;  [11] foreach_1.5.2               yulab.utils_0.0.5          </span></span>
<span><span class="co">#&gt;  [13] htmltools_0.5.2             fansi_1.0.3                </span></span>
<span><span class="co">#&gt;  [15] memoise_2.0.1               cluster_2.1.3              </span></span>
<span><span class="co">#&gt;  [17] doParallel_1.0.17           tzdb_0.3.0                 </span></span>
<span><span class="co">#&gt;  [19] openxlsx_4.2.5              limma_3.52.2               </span></span>
<span><span class="co">#&gt;  [21] ComplexHeatmap_2.12.0       modelr_0.1.8               </span></span>
<span><span class="co">#&gt;  [23] vroom_1.5.7                 pkgdown_2.0.5              </span></span>
<span><span class="co">#&gt;  [25] colorspace_2.0-3            rvest_1.0.2                </span></span>
<span><span class="co">#&gt;  [27] textshaping_0.3.6           haven_2.5.0                </span></span>
<span><span class="co">#&gt;  [29] xfun_0.31                   crayon_1.5.1               </span></span>
<span><span class="co">#&gt;  [31] RCurl_1.98-1.7              jsonlite_1.8.0             </span></span>
<span><span class="co">#&gt;  [33] impute_1.70.0               iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [35] glue_1.6.2                  gtable_0.3.0               </span></span>
<span><span class="co">#&gt;  [37] zlibbioc_1.42.0             XVector_0.36.0             </span></span>
<span><span class="co">#&gt;  [39] GetoptLong_1.0.5            DelayedArray_0.22.0        </span></span>
<span><span class="co">#&gt;  [41] shape_1.4.6                 MetaboCoreUtils_1.4.0      </span></span>
<span><span class="co">#&gt;  [43] scales_1.2.0                vsn_3.64.0                 </span></span>
<span><span class="co">#&gt;  [45] DBI_1.1.3                   Rcpp_1.0.8.3               </span></span>
<span><span class="co">#&gt;  [47] mzR_2.30.0                  viridisLite_0.4.0          </span></span>
<span><span class="co">#&gt;  [49] clue_0.3-61                 gridGraphics_0.5-1         </span></span>
<span><span class="co">#&gt;  [51] bit_4.0.4                   preprocessCore_1.58.0      </span></span>
<span><span class="co">#&gt;  [53] MsCoreUtils_1.8.0           htmlwidgets_1.5.4          </span></span>
<span><span class="co">#&gt;  [55] httr_1.4.3                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [57] ellipsis_0.3.2              pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [59] XML_3.99-0.10               sass_0.4.1                 </span></span>
<span><span class="co">#&gt;  [61] dbplyr_2.2.1                utf8_1.2.2                 </span></span>
<span><span class="co">#&gt;  [63] ggplotify_0.1.0             tidyselect_1.1.2           </span></span>
<span><span class="co">#&gt;  [65] rlang_1.0.3                 munsell_0.5.0              </span></span>
<span><span class="co">#&gt;  [67] cellranger_1.1.0            tools_4.2.1                </span></span>
<span><span class="co">#&gt;  [69] cachem_1.0.6                cli_3.3.0                  </span></span>
<span><span class="co">#&gt;  [71] QFeatures_1.6.0             generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [73] broom_1.0.0                 evaluate_0.15              </span></span>
<span><span class="co">#&gt;  [75] fastmap_1.1.0               mzID_1.34.0                </span></span>
<span><span class="co">#&gt;  [77] yaml_2.3.5                  ragg_1.2.2                 </span></span>
<span><span class="co">#&gt;  [79] bit64_4.0.5                 knitr_1.39                 </span></span>
<span><span class="co">#&gt;  [81] fs_1.5.2                    zip_2.2.0                  </span></span>
<span><span class="co">#&gt;  [83] AnnotationFilter_1.20.0     ncdf4_1.19                 </span></span>
<span><span class="co">#&gt;  [85] pbapply_1.5-0               xml2_1.3.3                 </span></span>
<span><span class="co">#&gt;  [87] compiler_4.2.1              rstudioapi_0.13            </span></span>
<span><span class="co">#&gt;  [89] plotly_4.10.0               png_0.1-7                  </span></span>
<span><span class="co">#&gt;  [91] affyio_1.66.0               reprex_2.0.1               </span></span>
<span><span class="co">#&gt;  [93] bslib_0.3.1                 stringi_1.7.6              </span></span>
<span><span class="co">#&gt;  [95] desc_1.4.1                  MSnbase_2.22.0             </span></span>
<span><span class="co">#&gt;  [97] lattice_0.20-45             ProtGenerics_1.28.0        </span></span>
<span><span class="co">#&gt;  [99] Matrix_1.4-1                ggsci_2.9                  </span></span>
<span><span class="co">#&gt; [101] vctrs_0.4.1                 pillar_1.7.0               </span></span>
<span><span class="co">#&gt; [103] lifecycle_1.0.1             jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [105] MALDIquant_1.21             GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [107] data.table_1.14.2           bitops_1.0-7               </span></span>
<span><span class="co">#&gt; [109] R6_2.5.1                    pcaMethods_1.88.0          </span></span>
<span><span class="co">#&gt; [111] affy_1.74.0                 codetools_0.2-18           </span></span>
<span><span class="co">#&gt; [113] MASS_7.3-57                 assertthat_0.2.1           </span></span>
<span><span class="co">#&gt; [115] rprojroot_2.0.3             rjson_0.2.21               </span></span>
<span><span class="co">#&gt; [117] withr_2.5.0                 GenomeInfoDbData_1.2.8     </span></span>
<span><span class="co">#&gt; [119] MultiAssayExperiment_1.22.0 parallel_4.2.1             </span></span>
<span><span class="co">#&gt; [121] hms_1.1.1                   grid_4.2.1                 </span></span>
<span><span class="co">#&gt; [123] rmarkdown_2.14              Spectra_1.6.0              </span></span>
<span><span class="co">#&gt; [125] lubridate_1.8.0</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>massdataset Developed by Xiaotao Shen. <br> from <i class="fa fa-university"></i> Stanford University <br><i class="fa fa-clinic-medical"></i>School of Medicine <br><i class="fa fa-dna"></i> Genetics Department</p>
</div>

<div class="pkgdown">
  <p></p>
<p><b>Contact Us:</b> <br><i class="fa fa-envelope"></i> <a href="mailto:shenxt1990@outlook.com" class="email">shenxt1990@outlook.com</a> <br><i class="fa fa-weixin"></i> <a href="https://www.shenxt.info/files/wechat_QR.jpg" class="external-link">shenxt1990</a> <br><i class="fa fa-network-wired"></i> <a href="https://www.tidymass.org/" class="external-link">A part of tidymass</a></p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '21ff2860978c5d2ceb7d4150364070ff',
    indexName: 'massdataset',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
