<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interoperation with other tools • massdataset</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="Interoperation with other tools">
<meta property="og:description" content="massdataset">
<meta property="og:image" content="https://tidymass.github.io/massdataset/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-000000-01"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-000000-01');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">massdataset</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Help documents
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    </li>
<li class="dropdown-header">Installation</li>
    <li>
      <a href="../articles/massdataset_install.html">Install massdataset</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">mass_dataset-class object</li>
    <li>
      <a href="../articles/data_import_and_export.html">Data import and export</a>
    </li>
    <li>
      <a href="../articles/mutate_ms2.html">Add MS2 spectra data into mass_dataset class object</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Information from mass_dataset</li>
    <li>
      <a href="../articles/extract_data.html">Extract data from mass_dataset</a>
    </li>
    <li>
      <a href="../articles/process_info.html">Processing information</a>
    </li>
    <li>
      <a href="../articles/html_process_info.html">Output html processing information</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Processing of mass_dataset</li>
    <li>
      <a href="../articles/summary_info.html">Summary information</a>
    </li>
    <li>
      <a href="../articles/add_information.html">Add new information to mass_dataset</a>
    </li>
    <li>
      <a href="../articles/filtering.html">Filtering samples/variables</a>
    </li>
    <li>
      <a href="../articles/accessor.html">Accessor</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Merge and split mass_dataset</li>
    <li>
      <a href="../articles/cbind_rbind_mass_dataset.html">rbind or cbind mass_dataset</a>
    </li>
    <li>
      <a href="../articles/merge_two_mass_dataset.html">Merge two mass_dataset</a>
    </li>
    <li>
      <a href="../articles/split_mass_dataset.html">Split mass_dataset</a>
    </li>
    <li class="divider">
    </li>
<li class="dropdown-header">Others</li>
    <li>
      <a href="../articles/tidyverse_verse.html">Process mass_dataset using tidyverse functions</a>
    </li>
    <li>
      <a href="../articles/base_function.html">R base functions for mass_dataset</a>
    </li>
    <li>
      <a href="../articles/ggplot_mass_dataset.html">mass_dataset and ggplot2</a>
    </li>
    <li>
      <a href="../articles/based_on_mass_dataset.html">Develop functions based on mass_dataset</a>
    </li>
    <li>
      <a href="../articles/interoperability_with_other_tools.html">Interoperation with other tools</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    News
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="dropdown-header">Blog posts</li>
    <li>
      <a href="../articles/massdataset.html">Version 0.4.0</a>
    </li>
    <li class="divider">
    </li>
<li>
      <a href="../news/index.html">Change log</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/contact_us.html">Contact us</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/tidymass/massdataset/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/xiaotaoshen1990" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interoperation with other tools</h1>
                        <h4 data-toc-skip class="author">Xiaotao Shen
(<a href="https://www.shenxt.info/" class="external-link uri">https://www.shenxt.info/</a>)</h4>
                        
            <h4 data-toc-skip class="date">Created on 2022-05-16 and
updated on 2022-05-16</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/tidymass/massdataset/blob/HEAD/vignettes/interoperability_with_other_tools.Rmd" class="external-link"><code>vignettes/interoperability_with_other_tools.Rmd</code></a></small>
      <div class="hidden name"><code>interoperability_with_other_tools.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="introduction">
<strong>Introduction</strong><a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>To make <code>tidyMass</code> and <code>massdataset</code> is more
interoperability with other toolts which have beed developed for omics
data processing and analysis, we provide several functions that could
convert “mass_dataset” and data formats that required by other tools. In
the meanwhile, functions that convert other data formats to
<code>mass_dataset</code> are also provided.</p>
</div>
<div class="section level1">
<h1 id="metdna">
<strong>MetDNA</strong><a class="anchor" aria-label="anchor" href="#metdna"></a>
</h1>
<p><code>MetDNA</code> is a web-based tool for metabolite annotation
using metabolic reaction network (<a href="http://metdna.zhulab.cn/" class="external-link uri">http://metdna.zhulab.cn/</a>). Users can eaisy convert
<code>mass_dataset</code> to files that required for
<code>MetDNA</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tidymass/massdataset" class="external-link">massdataset</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"expression_data"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sample_info"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sample_info_note"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"variable_info"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"variable_info_note"</span><span class="op">)</span>

<span class="va">object</span> <span class="op">=</span>
  <span class="fu"><a href="../reference/create_mass_dataset.html">create_mass_dataset</a></span><span class="op">(</span>
    expression_data <span class="op">=</span> <span class="va">expression_data</span>,
    sample_info <span class="op">=</span> <span class="va">sample_info</span>,
    variable_info <span class="op">=</span> <span class="va">variable_info</span>,
    sample_info_note <span class="op">=</span> <span class="va">sample_info_note</span>,
    variable_info_note <span class="op">=</span> <span class="va">variable_info_note</span>
  <span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">object</span>
<span class="co">#&gt; <span style="color: #BBBB00;">-------------------- </span></span>
<span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">massdataset version: 1.0.5 </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span><span style="color: #BBBB00;">-------------------- </span></span>
<span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">1.expression_data:</span>[ 1000 x 8 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">2.sample_info:</span>[ 8 x 4 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">3.variable_info:</span>[ 1000 x 3 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">4.sample_info_note:</span>[ 4 x 2 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">5.variable_info_note:</span>[ 3 x 2 data.frame]</span>
<span class="co">#&gt; <span style="color: #00BB00;">6.ms2_data:</span>[ 0 variables x 0 MS2 spectra]</span>
<span class="co">#&gt; <span style="color: #BBBB00;">-------------------- </span></span>
<span class="co"><span style="color: #BBBB00;">#&gt; </span><span style="color: #00BB00;">Processing information (extract_process_info())</span></span>
<span class="co"><span style="color: #00BB00;">#&gt; create_mass_dataset ---------- </span></span>
<span class="co"><span style="color: #00BB00;">#&gt; </span>      Package         Function.used                Time</span>
<span class="co">#&gt; 1 massdataset create_mass_dataset() 2022-05-16 13:59:40</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/export_mass_dataset4metdna.html">export_mass_dataset4metdna</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span>, 
                           path <span class="op">=</span> <span class="st">"convert/metdna"</span><span class="op">)</span>
<span class="co">#&gt; NULL</span></code></pre></div>
<p>The files will be exported in the folder “convert/metdna”.</p>
<p><img src="../reference/figures/Screen-Shot.png"></p>
<p>Peak table.</p>
<p><img src="../reference/figures/Screen-Shot%201.png"> sample_info.</p>
<p><img src="../reference/figures/Screen-Shot%202.png"></p>
</div>
<div class="section level1">
<h1 id="summarizedexperiment">
<strong>SummarizedExperiment</strong><a class="anchor" aria-label="anchor" href="#summarizedexperiment"></a>
</h1>
<p>The <code>SummarizedExperiment</code> class is used to store
rectangular matrices of experimental results, which are commonly
produced by sequencing and microarray experiments. This data structure
is supported by lots of tools in omics files in R environment. We can
use the <code>convert_mass_dataset2summarizedexperiment</code> function
to convert <code>mass_dataset</code> to
<code>SummarizedExperiment</code> class.</p>
<p>Please install <code>SummarizedExperiment</code> first.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">BiocManager</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SummarizedExperiment"</span><span class="op">)</span>
<span class="op">}</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span>
<span class="va">se_object</span> <span class="op">&lt;-</span> 
<span class="fu"><a href="../reference/convert_mass_dataset2summarizedexperiment.html">convert_mass_dataset2summarizedexperiment</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span>
<span class="va">se_object</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 1000 8 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(1000): M136T55_2_POS M79T35_POS ... M232T937_POS M301T277_POS</span>
<span class="co">#&gt; rowData names(3): variable_id mz rt</span>
<span class="co">#&gt; colnames(8): Blank_3 Blank_4 ... PS4P3 PS4P4</span>
<span class="co">#&gt; colData names(4): sample_id injection.order class group</span></code></pre></div>
</div>
<div class="section level1">
<h1 id="mztab-m-format">
<strong>mzTab-m format</strong><a class="anchor" aria-label="anchor" href="#mztab-m-format"></a>
</h1>
<p><code>mzTab-M</code> is a data standard for sharing quantitative
results in mass spectrometry metabolomics, which is also supported by
lots of tools in metabolomics/proteomics filed (<a href="https://pubs.acs.org/doi/10.1021/acs.analchem.8b04310" class="external-link uri">https://pubs.acs.org/doi/10.1021/acs.analchem.8b04310</a>).
In <code>massdataset</code>, we also provide two function to convert
<code>mass_data</code> class and <code>mzTab-m</code>.</p>
<div class="section level2">
<h2 id="convert-mass_dataset-to-mztab-m">Convert <code>mass_dataset</code> to <code>mzTab-M</code><a class="anchor" aria-label="anchor" href="#convert-mass_dataset-to-mztab-m"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/convert_mass_dataset2mztab.html">convert_mass_dataset2mztab</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">object</span>, 
                           path <span class="op">=</span> <span class="st">"convert/mztab"</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>The data is put in the folder “convert/mztab”. You can open it with
Excel.</p>
<p><img src="../reference/figures/Screen-Shot%203.png"></p>
</div>
</div>
<div class="section level1">
<h1 id="session-information">
<strong>Session information</strong><a class="anchor" aria-label="anchor" href="#session-information"></a>
</h1>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Big Sur 10.16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] SummarizedExperiment_1.22.0 Biobase_2.52.0             </span>
<span class="co">#&gt;  [3] GenomicRanges_1.44.0        GenomeInfoDb_1.28.1        </span>
<span class="co">#&gt;  [5] IRanges_2.26.0              S4Vectors_0.30.0           </span>
<span class="co">#&gt;  [7] BiocGenerics_0.38.0         MatrixGenerics_1.4.2       </span>
<span class="co">#&gt;  [9] matrixStats_0.60.0          BiocManager_1.30.16        </span>
<span class="co">#&gt; [11] forcats_0.5.1               stringr_1.4.0              </span>
<span class="co">#&gt; [13] dplyr_1.0.7                 purrr_0.3.4                </span>
<span class="co">#&gt; [15] readr_2.0.0                 tidyr_1.1.3                </span>
<span class="co">#&gt; [17] tibble_3.1.3                tidyverse_1.3.1            </span>
<span class="co">#&gt; [19] ggplot2_3.3.5               magrittr_2.0.1             </span>
<span class="co">#&gt; [21] masstools_0.99.13           massdataset_1.0.5          </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] readxl_1.3.1           backports_1.2.1        circlize_0.4.14       </span>
<span class="co">#&gt;   [4] systemfonts_1.0.2      plyr_1.8.6             lazyeval_0.2.2        </span>
<span class="co">#&gt;   [7] BiocParallel_1.26.1    Rdisop_1.52.0          digest_0.6.27         </span>
<span class="co">#&gt;  [10] foreach_1.5.1          htmltools_0.5.2        fansi_0.5.0           </span>
<span class="co">#&gt;  [13] memoise_2.0.0          cluster_2.1.2          doParallel_1.0.16     </span>
<span class="co">#&gt;  [16] tzdb_0.1.2             openxlsx_4.2.4         limma_3.48.1          </span>
<span class="co">#&gt;  [19] ComplexHeatmap_2.8.0   modelr_0.1.8           vroom_1.5.3           </span>
<span class="co">#&gt;  [22] pkgdown_2.0.1          colorspace_2.0-2       rvest_1.0.1           </span>
<span class="co">#&gt;  [25] textshaping_0.3.6      haven_2.4.1            xfun_0.31             </span>
<span class="co">#&gt;  [28] crayon_1.4.1           RCurl_1.98-1.3         jsonlite_1.7.2        </span>
<span class="co">#&gt;  [31] impute_1.66.0          iterators_1.0.13       glue_1.4.2            </span>
<span class="co">#&gt;  [34] gtable_0.3.0           zlibbioc_1.38.0        XVector_0.32.0        </span>
<span class="co">#&gt;  [37] GetoptLong_1.0.5       DelayedArray_0.18.0    shape_1.4.6           </span>
<span class="co">#&gt;  [40] scales_1.1.1           vsn_3.60.0             DBI_1.1.1             </span>
<span class="co">#&gt;  [43] Rcpp_1.0.7             mzR_2.26.1             viridisLite_0.4.0     </span>
<span class="co">#&gt;  [46] clue_0.3-59            gridGraphics_0.5-1     bit_4.0.4             </span>
<span class="co">#&gt;  [49] preprocessCore_1.54.0  MsCoreUtils_1.4.0      htmlwidgets_1.5.3     </span>
<span class="co">#&gt;  [52] httr_1.4.2             RColorBrewer_1.1-2     ellipsis_0.3.2        </span>
<span class="co">#&gt;  [55] pkgconfig_2.0.3        XML_3.99-0.6           sass_0.4.0            </span>
<span class="co">#&gt;  [58] dbplyr_2.1.1           utf8_1.2.2             ggplotify_0.0.8       </span>
<span class="co">#&gt;  [61] tidyselect_1.1.1       rlang_0.4.11           munsell_0.5.0         </span>
<span class="co">#&gt;  [64] cellranger_1.1.0       tools_4.1.2            cachem_1.0.5          </span>
<span class="co">#&gt;  [67] cli_3.0.1              generics_0.1.0         broom_0.7.9           </span>
<span class="co">#&gt;  [70] evaluate_0.14          fastmap_1.1.0          mzID_1.30.0           </span>
<span class="co">#&gt;  [73] yaml_2.2.1             ragg_1.1.3             bit64_4.0.5           </span>
<span class="co">#&gt;  [76] knitr_1.33             fs_1.5.0               zip_2.2.0             </span>
<span class="co">#&gt;  [79] ncdf4_1.17             pbapply_1.4-3          xml2_1.3.2            </span>
<span class="co">#&gt;  [82] compiler_4.1.2         rstudioapi_0.13        plotly_4.9.4.1        </span>
<span class="co">#&gt;  [85] png_0.1-7              affyio_1.62.0          reprex_2.0.0          </span>
<span class="co">#&gt;  [88] bslib_0.3.1            stringi_1.7.3          desc_1.3.0            </span>
<span class="co">#&gt;  [91] MSnbase_2.18.0         lattice_0.20-45        ProtGenerics_1.24.0   </span>
<span class="co">#&gt;  [94] Matrix_1.3-4           ggsci_2.9              vctrs_0.3.8           </span>
<span class="co">#&gt;  [97] pillar_1.6.2           lifecycle_1.0.0        jquerylib_0.1.4       </span>
<span class="co">#&gt; [100] MALDIquant_1.20        GlobalOptions_0.1.2    data.table_1.14.0     </span>
<span class="co">#&gt; [103] bitops_1.0-7           R6_2.5.0               pcaMethods_1.84.0     </span>
<span class="co">#&gt; [106] affy_1.70.0            codetools_0.2-18       MASS_7.3-54           </span>
<span class="co">#&gt; [109] assertthat_0.2.1       rprojroot_2.0.2        rjson_0.2.20          </span>
<span class="co">#&gt; [112] withr_2.4.2            GenomeInfoDbData_1.2.6 hms_1.1.0             </span>
<span class="co">#&gt; [115] grid_4.1.2             rmarkdown_2.14         rvcheck_0.1.8         </span>
<span class="co">#&gt; [118] Cairo_1.5-12.2         lubridate_1.7.10</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>massdataset Developed by Xiaotao Shen. <br> from
<i class="fa fa-university"></i> Stanford University <br><i class="fa fa-clinic-medical"></i>School of Medicine <br><i class="fa fa-dna"></i> Genetics Department</p>
</div>

<div class="pkgdown">
  <p></p>
<p><b>Contact Us:</b> <br><i class="fa fa-envelope"></i> <a href="mailto:shenxt1990@outlook.com" class="email">shenxt1990@outlook.com</a> <br><i class="fa fa-weixin"></i> <a href="https://www.shenxt.info/files/wechat_QR.jpg" class="external-link">shenxt1990</a> <br><i class="fa fa-network-wired"></i> <a href="https://www.tidymass.org/" class="external-link">A part of tidymass</a></p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '21ff2860978c5d2ceb7d4150364070ff',
    indexName: 'massdataset',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
